src/components/AutoComplete/AutoComplete.tsx

```diff
@@ -198,28 +212,26 @@ function AutoComplete<Item extends ListItemKey = ListItemKey>(
           {...getMenuProps()}
           css={theme.autoComplete.getCSS({
             targetElement: "ul",
-            isOpen: menuIsOpen,
+            isOpen: showClearIcon,
           })}
         >
-          {menuIsOpen && (
-            <>
-              {items.map((record, index) => (
-                <li
-                  key={record.id || index}
-                  css={theme.autoComplete.getCSS({
-                    targetElement: "li",
-                    isHighlighted:
-                      highlightedIndex === index ||
-                      __internal__highlightedIndex === index,
-                  })}
-                  {...getItemProps({ index, item: record })}
-                >
-                  {renderListItem(record)}
-                </li>
-              ))}
-              {Footer && <Footer closeMenu={closeMenu} />}
-            </>
-          )}
+          {showClearIcon &&
+            items.map((record, index) => (
+              <li
+                key={record.id || index}
+                css={theme.autoComplete.getCSS({
+                  targetElement: "li",
+                  isHighlighted:
+                    highlightedIndex === index ||
+                    __internal__highlightedIndex === index,
+                })}
+                {...getItemProps({ index, item: record })}
+              >
+                {index === items.length - 1 && !!renderItemsFooter
```

Comment:
```suggestion
                {index === items.length - 1 && hasFooter
```

Reply:
Unfortunately cannot :( just because TS cannot pick up the correct type in the block scope. I wished I could :D